### ISSUE TITLE
[ASK] How to use tensorboard for logging NRMS trainning loss?

### ISSUE THREAD

**Issue by thinkall** [microsoft/recommenders#1329]
I'm trying to use the NRMS model, it runs well. The thing is that I can't find a way to integrate tensorboard for logging.

Any help is appreciated.

Thanks

**Comment by thinkall** [comment]
Problem solved. My solution is as below.

Just need to add some code in the ```fit``` function in ```base_model.py```.
```
......
        Returns:
            obj: An instance of self.
        """

        # below added for tensorboard
        self.writer = tf.compat.v1.summary.FileWriter('./logs', graph=tf.get_default_graph())
        tb_total_loss = tf.Variable(0, dtype=tf.float32)
        tb_data_loss = tf.Variable(0, dtype=tf.float32)
        summ_total_loss = tf.compat.v1.summary.scalar('total_loss', tb_total_loss)
        summ_data_loss = tf.compat.v1.summary.scalar('data_loss', tb_data_loss)
        sess = tf.Session()

        for epoch in range(1, self.hparams.epochs + 1):
            step = 0
            self.hparams.current_epoch = epoch

...other code...

            for batch_data_input in tqdm_util:

                step_result = self.train(batch_data_input)
                step_data_loss = step_result

                epoch_loss += step_data_loss
                step += 1
                if step % self.hparams.show_step == 0:
                    tqdm_util.set_description(
                        "step {0:d} , total_loss: {1:.4f}, data_loss: {2:.4f}".format(
                            step, epoch_loss / step, step_data_loss
                        )
                    )

                # added for tensorboard
                sess.run(tb_total_loss.assign(epoch_loss / step))
                sess.run(tb_data_loss.assign(step_data_loss))
                merged = tf.compat.v1.summary.merge([summ_total_loss, summ_data_loss])
                self.writer.add_summary(sess.run(merged), step)

            train_end = time.time()
            train_time = train_end - train_start

...other code...
```
